### DISCUSSION TITLE
Secure storeing of token and secrets

### DISCUSSION THREAD

**Discussion by TmaAlexAndersen** [vercel/next.js#23347] [Help] [answered]
We are facing the security issue of having a React SPA, Backend API and an IdentityServer, in our case AAD B2C. To our knowledge a SPA would normally store token and secrets in the browser, in order to request the API. Does the introduction of "Server side rendering", comply with the use case of tokens and secrets never being exposed/stored in the browser?

**Comment by akellbl4** [comment, marked as answer]
I've never worked with AAD B2C. Usually, the frontend needs only a token with `httpOnly` for having auth. Maybe it's some way to do it like that.
If we talk about Next.js you can put your tokens in the environment with `.env` and they'll be available only on the server-side. Only vars with `NEXT_PUBLIC_` prefix are exposed to client-side code.

**Reply by TmaAlexAndersen** [reply]
thank you for clearing this for me.
